main.go

```diff
@@ -22,40 +22,49 @@ type E3DBHTTPAuthorizer interface {
 	AuthHTTPClient(ctx context.Context) *http.Client
 }
 
-// HTTPError wraps details of an HTTP error
-type HTTPError struct {
+// RequestError provides additional details about the failed request.
+type RequestError struct {
 	message    string
 	URL        string
 	StatusCode int
 }
 
-// Error implements the error interface for HTTPError.
-func (err *HTTPError) Error() string {
+// Error implements the error interface for RequestError.
+func (err *RequestError) Error() string {
 	return err.message
 }
 
+// NewError creates a new RequestError
+func NewError(message, url string, statusCode int) *RequestError {
```

Comment:
small: `NewRequestError` in case we want additional error types?

Reply:
Worth doing, nice call out.